tests/integrationv2/providers.py

```diff
@@ -225,21 +218,25 @@ def _join_ciphers(self, ciphers):
     def _cipher_to_cmdline(self, protocol, cipher):
         cmdline = list()
 
-        if cipher.min_version is Protocols.TLS13:
-            cmdline.append('-ciphersuites')
-        else:
-            cmdline.append('-cipher')
-
         ciphers = []
         if type(cipher) is list:
+            version = cipher[0].min_version
+
+            # Verify that each cipher in the list is the same version to prevent
+            # errors on the command line
+            mismatched_version = [x.min_version for x in cipher if x.min_version != version]
+            if len(mismatched_version) > 0:
+                raise Exception("Cipher string contains mismatched versions: {}".format([c.name for c in cipher]))
```

Comment:
Aren't different min_versions allowed for <TLS1.3? The only restriction is between pre-TLS1.3 and TLS1.3

Also, this check has been wrong a couple times now. Can we test it somehow?